spec/models/chargeback_spec.rb

```diff
@@ -0,0 +1,62 @@
+describe Chargeback do
+  let(:rpt) { MiqReport.new(:db => chargeback_model.name, :headers => [], :col_formats => []) }
+  let(:chargeback_model) { ChargebackVm }
+
+  # called from ReportController::Reports::Editor
+  # must be called on a subclass of Chargeback
+  describe ".set_chargeback_report_options" do
+    context "with ChargebackVm" do
+      let(:chargeback_model) { ChargebackVm }
+      it "groups by date-only" do
+        chargeback_model.set_chargeback_report_options(rpt, "date-only", "Tag", "Group By", "UTF")
+
+        expect(rpt.cols).to eq(%w[start_date display_range])
+        expect(rpt.col_order).to eq(%w[display_range])
+        expect(rpt.sortby).to eq(%w[start_date])
+      end
+
+      it "groups by project" do
+        chargeback_model.set_chargeback_report_options(rpt, "project", "Tag", "Group By", "UTF")
+
+        expect(rpt.cols).to eq(%w[start_date display_range project_name])
+        expect(rpt.col_order).to eq(%w[project_name display_range])
+        expect(rpt.sortby).to eq(%w[project_name start_date])
+      end
```

Comment:
how can a ChargebackVm even have a project or be grouped by project? 

Since this one is unlikely to actually be able to occur since the user can't choose it I'm going to merge, but I feel like this particular case should be removed (or replaced with something VM specific)